subsys/usb/class/hid/core.c

```diff
@@ -149,11 +149,23 @@ static void hid_int_in(u8_t ep, enum usb_dc_ep_cb_status_code ep_status)
 	hid_device.ops->int_in_ready();
 }
 
+static void hid_int_out(u8_t ep, enum usb_dc_ep_cb_status_code ep_status)
+{
+	u8_t data[CONFIG_HID_INTERRUPT_EP_MPS];
+	int len = 0;
+
+	usb_read(ep, data, sizeof(data), &len);
+
+	hid_device.ops->int_out_read(data, len);
```

Comment:
what if there is no callback?

Reply:
ok. I will read the data only if call back is registered othewise  i will just return. Is that ok.